### PR TITLE
Replace UMC_ENABLE_* defines with MFX_ENABLE_*

### PR DESCRIPTION
**Pull request by IlyaSergeyev**
1) removed AUDIO defs
2) UMC defs changed to MFX
3) implemented minimal mfxconfig.h.in
4) several steps in moving definitions from mfx_config.h to mfxconfig.h.in/cmake options

### CODE REVIEW

File: _studio/shared/umc/core/umc/include/umc_defs.h [hunk 1]
```diff
@@ -34,22 +34,19 @@
...
 
     // video decoders
-    #define UMC_ENABLE_MJPEG_VIDEO_DECODER
-    #define UMC_ENABLE_VP8_VIDEO_DECODER
-    #define UMC_ENABLE_VC1_VIDEO_DECODER
-    #define UMC_ENABLE_H264_VIDEO_DECODER
-    #define UMC_ENABLE_H265_VIDEO_DECODER
-    #define UMC_ENABLE_MPEG2_VIDEO_DECODER
-    #define UMC_ENABLE_VP9_VIDEO_DECODER
+    #define MFX_ENABLE_VC1_VIDEO_DECODE
```
Comment (dvrogozh) [inline _studio/shared/umc/core/umc/include/umc_defs.h]: no. Please, don't define same thing twice. We have this in mfx_config.h. Thus, include it here. Actually this file should contain single line (and copyright of course):

`#include <mfx_config.h>`

File: _studio/hevc_fei/CMakeLists.txt [hunk 2]
```diff
@@ -94,8 +94,9 @@ set( sources.plus "" )
 set( USE_STRICT_NAME TRUE )
 set(MFX_LDFLAGS "${MFX_ORIG_LDFLAGS} -Wl,--version-script=${MFX_ROOTDIR}/mfx_lib/plugin/libmfxsw_plugin.map" )
 
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/../hevce_hw/h265/include )
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/h265_fei/include )
+include_directories( "${MSDK_STUDIO_ROOT}/hevce_hw/h265/include" )
```
Comment (dvrogozh) [inline _studio/hevc_fei/CMakeLists.txt]: why you need this in this? I think you need drop it.

File: _studio/hevce_hw/CMakeLists.txt [hunk 3]
```diff
@@ -47,8 +47,9 @@ set( sources.plus "" )
 set( LIBS "" )
 set( defs "" )
 
-include_directories( ${MSDK_STUDIO_ROOT}/hevce_hw/include )
-include_directories( ${MSDK_STUDIO_ROOT}/hevce_hw/h265/include )
+include_directories( "${MSDK_STUDIO_ROOT}/hevce_hw/h265/include" )
```
Comment (dvrogozh) [inline _studio/hevce_hw/CMakeLists.txt]: why you need this in this? I think you need drop it.

File: _studio/shared/umc/core/umc/include/umc_defs.h [hunk 4]
```diff
@@ -33,20 +33,7 @@
...
-    #define MFX_ENABLE_H265_VIDEO_DECODE
-    #define MFX_ENABLE_VP8_VIDEO_DECODE
-    #define MFX_ENABLE_VP9_VIDEO_DECODE
-
-    // video encoders
-    #define MFX_ENABLE_MVC_VIDEO_ENCODE
-    #define MFX_ENABLE_MJPEG_VIDEO_ENCODE
-    #define MFX_ENABLE_MPEG2_VIDEO_ENCODE
-    #define MFX_ENABLE_H264_VIDEO_ENCODE
+#include "mfx_config.h"
```
Comment (dvrogozh) [inline _studio/shared/umc/core/umc/include/umc_defs.h]: this is part of the previous patch! By the way, I very much doubt you need if-defined-LINUX-ANDROID...

File: CMakeLists.txt [hunk 5]
```diff
@@ -63,6 +63,9 @@ else()
   message( STATUS "There is no ConfTargets.cmake" )
 endif()
 
+set( MFX_CONFIG_FILE ${BUILDER_ROOT}/profiles/fastboot.cmake )
```
Comment (dvrogozh) [inline CMakeLists.txt]: no, no, no. Fastboot will _not_ be a default config! never!

Please, cherry-pick this patch: https://github.com/dvrogozh/MediaSDK/commit/01c44ad8ad50bba79294bfda22a20c3a8b899d70. This is a starting point! What you are doing is simply incorrect. In my original patch user can either add -DMFX_ENABLE_H264_VIDEO_DECODER=OFF options or have -DMFX_CONFIG_FILE=/path/to/file. Try it works. And this actually should be put into the commit message.

File: _studio/shared/include/mfx_config.h [hunk 6]
```diff
@@ -51,8 +52,8 @@
 #if !defined(LINUX_TARGET_PLATFORM) || defined(LINUX_TARGET_PLATFORM_BDW) || defined(LINUX_TARGET_PLATFORM_CFL) || defined(LINUX_TARGET_PLATFORM_BXT)
     #if !defined(ANDROID)
         // HW decoders are part of library
-        #define MFX_ENABLE_H264_VIDEO_DECODE
-        #define MFX_ENABLE_H265_VIDEO_DECODE
+        #define MFX_ENABLE_VP8_VIDEO_DECODE
```
Comment (dvrogozh) [inline _studio/shared/include/mfx_config.h]: Idea was to move MFX_ENABLE_H264_VIDEO_DECODE to mfxconfig.h.in, not to add some new ENABLE defines for other codecs!

File: builder/ConfCodecs.cmake [hunk 7]
```diff
@@ -27,4 +27,22 @@ endif()
...
+option ( MFX_ENABLE_H264_VIDEO_DECODE "Enabled H264 decoder?" ON)
+option ( MFX_ENABLE_H265_VIDEO_DECODE "Enabled H265 decoder?" ON)
+option ( MFX_ENABLE_VC1_VIDEO_DECODE "Enabled VC1 decoder?" ON)
+option ( MFX_ENABLE_MJPEG_VIDEO_DECODE "Enabled MJPEG decoder?" ON)
+option ( MFX_ENABLE_MPEG2_VIDEO_DECODE "Enabled MPEG2 decoder?" ON)
+option ( MFX_ENABLE_VP8_VIDEO_DECODE "Enabled VP8 decoder?" ON)
+option ( MFX_ENABLE_VP9_VIDEO_DECODE "Enabled VP9 decoder?" ON)
+
+option ( MFX_ENABLE_H264_VIDEO_ENCODE "Enabled H264 encoder?" ON)
+option ( MFX_ENABLE_H265_VIDEO_ENCODE "Enabled H265 encoder?" OFF)
```
Comment (dvrogozh) [inline builder/ConfCodecs.cmake]: They all should be enabled by default!

File: CMakeLists.txt [hunk 8]
```diff
@@ -63,7 +63,7 @@ else()
   message( STATUS "There is no ConfTargets.cmake" )
 endif()
 
-set( MFX_CONFIG_FILE ${BUILDER_ROOT}/profiles/fastboot.cmake )
+#set( MFX_CONFIG_FILE ${BUILDER_ROOT}/profiles/fastboot.cmake )
```
Comment (dvrogozh) [inline CMakeLists.txt]: this is simply not needed - remove

File: CMakeLists.txt [hunk 9]
```diff
@@ -63,6 +63,9 @@ else()
   message( STATUS "There is no ConfTargets.cmake" )
```
Comment (dvrogozh) [inline CMakeLists.txt]: please, change the above to unconditionally include ${BUILDER_ROOT}/ConfTargets.cmake

File: builder/ConfCodecs.cmake [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+if (DEFINED MFX_CONFIG_FILE)
```
Comment (dvrogozh) [inline builder/ConfCodecs.cmake]: I don't think we need a separate file for this. We have ConfTargets.cmake already,

I really want my change to be cherry-picked!

File: _studio/shared/include/mfx_config.h [hunk 11]
```diff
@@ -69,9 +65,7 @@
             #define MFX_ENABLE_H264_VIDEO_FEI_PAK
         #endif
 
-        // vpp
-        #define MFX_ENABLE_DENOISE_VIDEO_VPP
-        #define MFX_ENABLE_VPP
+        // ToDo: do something with it
```
Comment (dvrogozh) [inline _studio/shared/include/mfx_config.h]: let's check together how we will move enable defines to the mfxconfig.h.in

File: _studio/shared/include/mfx_config.h (hunk 6 above)
Comment (IlyaSergeyev) [reply _studio/shared/include/mfx_config.h]: There are compilation errors without MFX_ENABLE_VP... These rows are moved from umc_defs.h

File: _studio/hevce_hw/CMakeLists.txt (hunk 3 above)
Comment (IlyaSergeyev) [reply _studio/hevce_hw/CMakeLists.txt]: There are no folder hwce_hw/include, but there is hevce_hw/h265/include.

File: _studio/shared/umc/core/umc/include/umc_defs.h (hunk 4 above)
Comment (IlyaSergeyev) [reply _studio/shared/umc/core/umc/include/umc_defs.h]: And we had 2 different sets of android defs! First in umc_defs.h and second in mfx_android_defs.h!

**Comment by IlyaSergeyev** [comment]
Rebased to commit bd7e5a5

**Comment by dvrogozh** [comment]
@onabiull : rebase done.

**Comment by IlyaSergeyev** [comment]
Closed, see #506 unstead

**Pull request closed without merging**